cdap-api/src/main/java/co/cask/cdap/api/schedule/ScheduleBuilder.java

```diff
@@ -177,34 +158,58 @@
 
   /**
    * Creates a schedule which is triggered in the same application version.
    *
-   * @see #setTriggerOnProgramStatus(String, String, ProgramType, String, ProgramStatus...)
+   * @see #triggerOnProgramStatus(String, String, ProgramType, String, ProgramStatus...)
    */
-  ScheduleCreationBuilder setTriggerOnProgramStatus(String programNamespace, String application,
-                                                    ProgramType programType, String program,
-                                                    ProgramStatus... programStatuses);
+  ScheduleCreationSpec triggerOnProgramStatus(String programNamespace, String application, ProgramType programType,
+                                              String program, ProgramStatus... programStatuses);
 
   /**
    * Creates a schedule which is triggered in the same application and application version.
    *
-   * @see #setTriggerOnProgramStatus(String, String, ProgramType, String, ProgramStatus...)
+   * @see #triggerOnProgramStatus(String, String, ProgramType, String, ProgramStatus...)
    */
-  ScheduleCreationBuilder setTriggerOnProgramStatus(String programNamespace, ProgramType programType, String program,
-                                                    ProgramStatus... programStatuses);
+  ScheduleCreationSpec triggerOnProgramStatus(String programNamespace, ProgramType programType, String program,
+                                              ProgramStatus... programStatuses);
 
   /**
    * Creates a schedule which is triggered in the same namespace, application, and application version.
    *
-   * @see #setTriggerOnProgramStatus(String, String, ProgramType, String, ProgramStatus...)
+   * @see #triggerOnProgramStatus(String, String, ProgramType, String, ProgramStatus...)
    */
-  ScheduleCreationBuilder setTriggerOnProgramStatus(ProgramType programType, String program,
-                                                    ProgramStatus... programStatuses);
+  ScheduleCreationSpec triggerOnProgramStatus(ProgramType programType, String program,
+                                              ProgramStatus... programStatuses);
+
+  /**
+   * Inner class that creates a ScheduleCreationSpec from the deployed properties
+   */
+  class ScheduleCreationBuilder extends ScheduleCreationSpec {
+    private final TriggerBuilder triggerBuilder;
+
+    public ScheduleCreationBuilder(String name, String description, String programName, Map<String, String> properties,
+                                   List<? extends Constraint> constraints, long timeoutMillis,
+                                   TriggerBuilder triggerBuilder) {
+      super(name, description, programName, properties, null, constraints, timeoutMillis);
+      this.triggerBuilder = triggerBuilder;
+    }
+
+    @Override
+    public Trigger getTrigger() {
+      throw new IllegalStateException(String.format("Schedule %s does not have a trigger", getName()));
```

Comment:
`IllegalStateException` seems not appropriate. There's no state change in this class. Also improve the error message to indicate that trigger has not been built because of missing infos.